x-pack/plugins/fleet/public/applications/fleet/components/fleet_server_instructions/hooks/use_fleet_server_host.ts

```diff
@@ -59,39 +58,46 @@ export const useFleetServerHost = (): FleetServerHostForm => {
     [hostUrlsInput, nameInput]
   );
 
-  const { data } = useGetFleetServerHosts();
+  const { data, resendRequest: refreshGetFleetServerHosts } = useGetFleetServerHosts();
+
+  const fleetServerHosts = useMemo(() => data?.items ?? [], [data?.items]);
+
+  useEffect(() => {
+    const defaultHost = fleetServerHosts.find((item) => item.is_default === true);
+    if (defaultHost) {
+      setFleetServerHost(defaultHost);
+    } else {
+      setFleetServerHost(null);
+    }
+  }, [fleetServerHosts]);
 
   const saveFleetServerHost = useCallback(
-    async (newFleetServerHost: FleetServerHost) => {
+    async (newFleetServerHost: Omit<FleetServerHost, 'id'>) => {
       setIsFleetServerHostSubmitted(false);
-      setFleetServerHost(newFleetServerHost);
 
-      const fleetServerHostExists = data?.items.reduce((acc, curr) => {
-        const hostsIntersection = intersection(curr.host_urls, newFleetServerHost?.host_urls);
-        return hostsIntersection.length > 0 || acc;
-      }, false);
-
-      // If the Fleet Server host provided already exists in settings, don't submit it
-      if (fleetServerHostExists) {
```

Comment:
I think that this was originally here to speed things up when the host already exists, but maybe you're right that it's not needed anymore